pybossa/repositories/task_repository.py

```diff
@@ -497,22 +495,34 @@ def update_priority(self, project_id, priority, filters):
         self.db.session.commit()
         cached_projects.clean_project(project_id)
 
-    def find_duplicate(self, project_id, info):
+    def find_duplicate(self, project_id, info, dup_checksum=None):
         """
         Find a task id in the given project with the project info using md5
         index on info column casted as text. Md5 is used to avoid key size
         limitations in BTree indices
         """
-        sql = text('''
-                   SELECT task.id as task_id
-                   FROM task
-                   WHERE task.project_id=:project_id
-                   AND task.state='ongoing'
-                   AND md5(task.info::text)=md5(((:info)::jsonb)::text)
-                   ''')
-        info = json.dumps(info, allow_nan=False)
-        row = self.db.session.execute(
-            sql, dict(info=info, project_id=project_id)).first()
+        # with task payload containing dup_checksum value, perform duplicate
+        # check based on checkum instead of comparing entire task payload info
+        if dup_checksum:
+            sql = text('''
+                    SELECT task.id as task_id
+                    FROM task
+                    WHERE task.project_id=:project_id
+                    AND task.dup_checksum=:dup_checksum
+                    ''')
+            row = self.db.session.execute(
+                sql, dict(project_id=project_id, dup_checksum=dup_checksum)).first()
+        else:
+            sql = text('''
+                    SELECT task.id as task_id
+                    FROM task
+                    WHERE task.project_id=:project_id
+                    AND task.state='ongoing'
+                    AND md5(task.info::text)=md5(((:info)::jsonb)::text)
```

Comment:
We might consider running a job to backfill checksums for existing tasks. This would boot performance to not have to enter this else branch which computes the hash.

Reply:
checksum would be generated for projects with `duplicate_task_check` configured.  The else block would do regular duplicate check as it was originally doing